### PR TITLE
QGA-8 Create user routes

### PR DESCRIPTION
**Pull request by nondante**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import index from './routes/index';
+import user from './routes/user';
```
Comment (Gruseckis) [inline server.js]: Make this more meaningful.  What is user? Also name for this route should be better maybe usersRouter?

File: models/userModel.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+const userSchema = new mongoose.Schema(
+{
+  username: {type: String, trim: true, unique: true, required: true},
+  email: {type: String, trim: true, required: true},
+  hashedPassword: {type: String, trim: true, required: true},
+  name: {type: String, trim: true, required: true},
+  surname: {type: String, trim: true, required: true},
+  dateOfBirth: {type: Date, trim: true, required: true},
+  level: {type: String, trim: true, required: true}
```
Comment (Gruseckis) [inline models/userModel.js]: This should be enum.

File: models/userModel.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    );
+  }
+  next();
+});
+
+const UserModel = mongoose.model('User',userSchema);
+
+const save = async model => new UserModel(model).save();
+
+const getUserByName = async username => UserModel.findOne({username});
```
Comment (Gruseckis) [inline models/userModel.js]: Incorrect name for function. You are getting user by username not name.

File: models/userModel.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const getUsers = async () => UserModel.find();
+const updateUser = async(id,model) => UserModel.findByIdAndUpdate(id, model, {new: true});
+const deleteUser = async(id) =>  UserModel.findByIdAndRemove(id);
+
+
```
Comment (Gruseckis) [inline models/userModel.js]: Please remove extra empty lines

File: models/userModel.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+const comparePassword = async ({userPassword, hashedPassword}) => bcrypt.compare(userPassword, hashedPassword);
+
+
```
Comment (Gruseckis) [inline models/userModel.js]: Please leave only one empty line if you need to

File: controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+
+import * as UserModel from "../models/userModel";
```
Comment (Gruseckis) [inline controllers/userController.js]: Why are you importing 8 functions when you only use 3. Please import only functions that you will use.

File: controllers/userController.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import AppError from "../errors/AppError";
+
+const logger = require('../utils/logger')('logController');
+
+const getUserInfo = async (req, res) => {
+  logger.log('debug', 'logIn: %j', req.body);
+  const { user } = req;
+  res.status(200).send({
+    payload: {
+      username: user.username,
```
Comment (Gruseckis) [inline controllers/userController.js]: you can just send user as payload

File: controllers/userController.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
+
+import * as UserModel from "../models/userModel";
+import AppError from "../errors/AppError";
+
+const logger = require('../utils/logger')('logController');
+
+const getUserInfo = async (req, res) => {
+  logger.log('debug', 'logIn: %j', req.body);
+  const { user } = req;
```
Comment (Gruseckis) [inline controllers/userController.js:9]: To be honest we do not have this object attached to req. So your code won'r work.

File: models/userModel.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import mongoose from 'mongoose';
```
Comment (Gruseckis) [inline models/userModel.js]: I would like you to remove this file from PR and make it in separate PR.

**Review by Gruseckis — CHANGES_REQUESTED** [review]

File: models/userModel.js (hunk 2 above)
Comment (KristineZelenivska) [reply models/userModel.js]: level by Default should be "user"
https://mongoosejs.com/docs/defaults.html

File: controllers/userController.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  res.status(200).send({
+    payload: {
+      user
+    },
+  });
+};
+
+const getAllUsers = async (req,res, next) => {
```
Comment (Gruseckis) [inline controllers/userController.js]: Please add space after comas

**Review by Gruseckis — CHANGES_REQUESTED** [review]

**Comment by Gruseckis** [comment]
change PR name to be the same as ticket number without spaces

**Review by Gruseckis — APPROVED** [review]

**Pull request merged**